src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -0,0 +1,100 @@
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.hints.HintInfo
+import com.intellij.codeInsight.hints.InlayInfo
+import com.intellij.codeInsight.hints.InlayParameterHintsProvider
+import com.intellij.codeInsight.hints.Option
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsCompositeElement
+import org.rust.lang.core.psi.ext.valueParameters
+import org.rust.lang.core.types.infer.inferDeclarationType
+import org.rust.lang.core.types.ty.TyUnit
+import org.rust.lang.core.types.ty.TyUnknown
+
+private val RsCallExpr.declaration: RsFunction?
+    get() = (expr as? RsPathExpr)?.path?.reference?.resolve() as? RsFunction
+
+private val RsMethodCallExpr.declaration: RsFunction?
+    get() = reference.resolve() as? RsFunction
+
+enum class HintType(desc: String, enabled: Boolean) {
+    LET_BINDING_HINT("Show local variable type hints", true) {
+        override fun provideHints(elem: PsiElement): List<InlayInfo> {
+            val element = elem as? RsLetDecl ?: return emptyList()
+            if (element.typeReference != null) return emptyList()
+            val ident = element.pat as? RsPatIdent ?: return emptyList()
+            val patBinding = ident.patBinding
+            val type = inferDeclarationType(patBinding)
+            if (type is TyUnknown || type is TyUnit) return emptyList()
```

Comment:
Is there a reason / use case why you do not want to show the unit type on local variables?
Noise during typing new code?

I think it might be useful to show it for situations like.
```rust
let z = accidentally_returns_nothing();

fn accidentally_returns_nothing() -> () {
    ;
}
```